src/libraries/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -68,6 +71,27 @@ public partial class Activity : IDisposable
         public object? GetCustomProperty(string propertyName) { throw null; }
         public ActivityContext Context { get { throw null; } }
     }
+    public class ActivityTagsCollection : System.Collections.Generic.IDictionary<string, object>
```

Comment:
Not `ActivityTagDictionary`? 🤷‍♂️

Reply:
No, because the collection has to keep the list in order. in general Dictionaries don't promise that.